File: java/src/test/java/topcoder/batchsystem/BatchSystemSpec.java
package topcoder.batchsystem;

import org.junit.Test;

import java.util.Arrays;

public class BatchSystemSpec {

  @Test
  public void test() {
    int[] durations = {400, 100, 100, 100};
    String[] users = {"Danny", "Stella", "Stella", "Mac"};

    BatchSystem batchSystem = new BatchSystem();
    int[] result = batchSystem.schedule(durations, users);
    System.out.println(Arrays.toString(result));

    int[] ans = {3, 1, 2, 0};
    assert(Arrays.equals(result, ans));
  }
}

File: scala/src/test/scala/topcoder/batchsystem/BatchSystemSpec.scala
package topcoder.batchsystem

import org.scalatest.{DiagrammedAssertions, FunSpec}

class BatchSystemSpec extends FunSpec with DiagrammedAssertions {
  describe("BatchSystem") {
    it("q1") {
      val durations = Array(400, 100, 100, 100)
      val users = Array("Danny", "Stella", "Stella", "Mac")
      val ans = Array(3, 1, 2, 0)

      val result = BatchSystem.schedule(durations, users)
      assert(ans.deep == result.deep)
    }
  }
}
